airflow/include/tasks/extract/utils/weaviate/ask_astro_weaviate_hook.py

```diff
@@ -220,84 +222,91 @@ def batch_ingest(
         :param uuid_column: Name of the column containing the UUID.
         :param vector_column: Name of the column containing the vector data.
         :param batch_params: Parameters for batch configuration.
-        :param existing: Strategy to handle existing data ('skip', 'replace', 'upsert').
-        :param verbose: Whether to print verbose output.
+        :param existing: Strategy to handle existing data ('skip', 'replace', 'upsert' or 'error').
+        :param verbose: Whether to log verbose output.
+        :param tenant: The tenant to which the object will be added.
         """
-        batch = self.client.batch.configure(**batch_params)
-        batch_errors = []
 
-        for row_id, row in df.iterrows():
-            data_object = row.to_dict()
-            uuid = data_object.pop(uuid_column)
-            vector = data_object.pop(vector_column, None)
-
-            try:
-                if self.client.data_object.exists(uuid=uuid, class_name=class_name) is True:
-                    if existing == "skip":
-                        if verbose is True:
-                            self.logger.warning(f"UUID {uuid} exists.  Skipping.")
+        # configuration for context manager for __exit__ method to callback on errors for weaviate batch ingestion.
+        if not batch_params.get("callback"):
+            batch_params.update({"callback": self.process_batch_errors})
+
+        self.client.batch.configure(**batch_params)
+
+        with self.client.batch as batch:
+            for row_id, row in df.iterrows():
+                data_object = row.to_dict()
+                uuid = data_object.pop(uuid_column)
+                vector = data_object.pop(vector_column, None)
+
+                try:
+                    if self.client.data_object.exists(uuid=uuid, class_name=class_name):
+                        if existing == "error":
+                            raise AirflowException(f"Ingest of UUID {uuid} failed.  Object exists.")
```

Comment:
```suggestion
                            raise AirflowException(f"Ingest of UUID {uuid} failed. Object exists.")

```

Reply:
add the additional line for separate the 2 if-else logic

Reply:
Done